test/integration/batcher/TwoWayBatcherIntegration.test.ts

```diff
@@ -0,0 +1,609 @@
+import { expect } from 'chai'
+import HRE from 'hardhat'
+import { Signer, utils, constants } from 'ethers'
+
+import { InstanceVars, deployProtocol } from '../setupHelpers'
+import { TwoWayBatcher, TwoWayBatcher__factory } from '../../../types/generated'
+import { impersonate } from '../../testutil'
+import { impersonateWithBalance } from '../../testutil/impersonate'
+
+const { ethers } = HRE
+
+const HARDCODED_RESERVE_BORROWER_ADDRESS = '0x0B663CeaCEF01f2f88EB7451C70Aa069f19dB997'
+
+describe('TwoWayBatcher', () => {
```

Comment:
can we add cases for a potentially malicious user transferring additional DSU / USDC into the contract? just to make sure nothing bricks.